client/components/email-verification/index.js

```diff
@@ -12,6 +13,20 @@ export default function emailVerification( context, next ) {
 	const showVerifiedNotice = '1' === context.query.verified;
 	const showNewEmailNotice = '1' === context.query.new_email_result;
 
+	// Redirect email verification to v2 for users who have opted into the new Hosting Dashboard
+	if (
+		showNewEmailNotice &&
+		( context.pathname === '/me/account' || context.pathname === '/settings/account' )
+	) {
+		const state = context.store.getState();
+		const hostingDashboardPreference = getPreference( state, 'hosting-dashboard-opt-in' );
+
+		if ( hostingDashboardPreference === 'opt-in' ) {
+			window.location.href = `/v2/me/profile?new_email_result=${ context.query.new_email_result }`;
```

Comment:
I'm not sure this is working. I tried by testing it on `/me/account?verified=1&new_email_result=1` and I wasn't redirected. I know you mentioned some backend changes might be required but I wonder if there's something else at play here.

Reply:
Yeah I wasn't having any success either but I think I figured it out! My state had the correct HD opt-in preference, but it wasn't picking it up for some reason (I had the correct `hostingDashboardPreference.value` when I was testing). Now that I wait for it to load first, the redirect works for me: [302dded](https://github.com/Automattic/wp-calypso/pull/106103/commits/302dded34dd9cb6c774ab8278932fea347001fdc)

For the backend changes, I meant that it'd be good to not have to play with this redirect stuff and just send the user to the correct route based on the v1/v2 version. At least the redirect is working for now!!

Side note: `/me/account?new_email_result=1` should work. I don't know if we have both params together.